samples/image_diff.cc

```diff
@@ -36,12 +36,24 @@ struct Options {
 };
 
 const char kUsage[] = R"(Usage: image_diff [options] image1.png image2.png
-
- options:
-  --rmse                    -- Compare using RMSE algorithm (default).
-  --histogram_emd           -- Compare using histogram EMD algorithm.
-  -t | --tolerance <float>  -- Tolerance value for comparison.
-  -h | --help               -- This help text.
+options:
+
+  --rmse TOLERANCE
+               Compare using the Root Mean Square Error (RMSE) algorithm with
+               a floating point TOLERANCE value in the range 0..255, where 0
+               indicates identical images and 255 indicates images where every
+               color channel has the maximum difference.
+
+  --histogram_emd TOLERANCE
```

Comment:
How is this going to work if there are multiple parameters? Listing them, something else?

Reply:
Not sure what you mean. Is there something preventing us to syntax below?

`--algorithm PARAM1 PARAM2`

I suppose if there were several parameters we would have to modify the layout.

Reply:
My concern was if this would get confusing. If I have a script which says:

`image_diff --foo 1 2 3 4 45 6 $1 $2` that's hard to read. Where:

`image_diff --foo --tolerance 1 --other 2 --thing 3 ...` is longer but potentially easier.

Reply:
Yes, there is a trade-off here. We want the parameters to be specific to each algorithm. `--tolerance` looks optional, it looks like it can go anywhere, and it looks like it applies globally. We want to be more specific. But the downside is as you say. However, this is a common issue with many conventional command line interfaces (and indeed C/C++ IMO). If we want to go against convention, we could match AmberScript: 

```
image_diff \
  "EMD_WINDOW TOLERANCE 7 WINDOW_SIZE 4" \
  "EMD TOLERANCE 7" \
  "RMSE TOLERANCE 7"
```

But I think losing some readability might be OK here since this is not a scripting language and our help is pretty clear.

Reply:
Another option:

```
image_diff \
  --histogram_emd_window=tolerance:7.0;window_size:4;
  --histogram_emd=tolerance:3.0;
  --rmse=tolerance:1.0;
```

Reply:
I assumed this would do one type of image comparison at a time, are you thinking you want it to be able to do multiple? My assumption was you'd do something like:

```
image_diff --rmse --tolerance 2.3
image_diff --histogram_emd --tolerance 1.4 --window-size 4
```

but you want to do both of them in one command? What's the output in that case, if rmse passes and histogram_emd fails, does image_diff pass or fail?

Reply:
Indeed, we don't need to run more than one algorithm. I think we agree to stick with the syntax in this PR, even though it may not always be obvious what the numbers mean.